NebulaPatcher/Patches/Dynamic/UIOptionWindow_Patch.cs

```diff
@@ -0,0 +1,279 @@
+        public static void _OnCreate_Postfix(UIOptionWindow __instance)
+        {
+            tempToUICallbacks = new Dictionary<string, System.Action>();
+            tempMultiplayerOptions = new MultiplayerOptions();
+
+            // Add multiplayer tab button
+            UIButton[] tabButtons = AccessTools.Field(__instance.GetType(), "tabButtons").GetValue(__instance) as UIButton[];
+            multiplayerTabIndex = tabButtons.Length;
+            RectTransform lastTab = tabButtons[tabButtons.Length - 1].GetComponent<RectTransform>();
+            RectTransform beforeLastTab = tabButtons[tabButtons.Length - 2].GetComponent<RectTransform>();
+            float tabOffset = lastTab.anchoredPosition.x - beforeLastTab.anchoredPosition.x;
+            multiplayerTab = Object.Instantiate(lastTab, lastTab.parent, true);
+            multiplayerTab.anchoredPosition = new Vector2(lastTab.anchoredPosition.x + tabOffset, lastTab.anchoredPosition.y);
+            UIButton[] newTabButtons = CollectionExtensions.AddToArray(tabButtons, multiplayerTab.GetComponent<UIButton>());
+            AccessTools.Field(__instance.GetType(), "tabButtons").SetValue(__instance, newTabButtons);
+
+            // Update multiplayer tab text
+            Text tabText = multiplayerTab.GetComponentInChildren<Text>();
+            tabText.GetComponent<Localizer>().enabled = false;
+            tabText.text = "Multiplayer";
+            Text[] tabTexts = AccessTools.Field(__instance.GetType(), "tabTexts").GetValue(__instance) as Text[];
+            Text[] newTabTexts = CollectionExtensions.AddToArray(tabTexts, tabText);
+            AccessTools.Field(__instance.GetType(), "tabTexts").SetValue(__instance, newTabTexts);
+
+            // Add multiplayer tab content
+            Tweener[] tabTweeners = AccessTools.Field(__instance.GetType(), "tabTweeners").GetValue(__instance) as Tweener[];
+            RectTransform contentTemplate = tabTweeners[0].GetComponent<RectTransform>();
+            multiplayerContent = Object.Instantiate(contentTemplate, contentTemplate.parent, true);
+            multiplayerContent.name = "multiplayer-content";
+
+            Tweener[] newContents = CollectionExtensions.AddToArray(tabTweeners, multiplayerContent.GetComponent<Tweener>());
+            AccessTools.Field(__instance.GetType(), "tabTweeners").SetValue(__instance, newContents);
+            UIButton[] revertButtons = AccessTools.Field(__instance.GetType(), "revertButtons").GetValue(__instance) as UIButton[];
+            RectTransform revertButton = multiplayerContent.Find("revert-button").GetComponent<RectTransform>();
+            UIButton[] newRevertButtons = CollectionExtensions.AddToArray(revertButtons, revertButton.GetComponent<UIButton>());
+            AccessTools.Field(__instance.GetType(), "revertButtons").SetValue(__instance, newRevertButtons);
+
+            // Find control templates
+            foreach (RectTransform child in multiplayerContent)
```

Comment:
Would probably be a bit simpler if you have a Panel holding onto all the multiplayerContent children that need removing.

Reply:
Then it becomes `Object.Destroy(panel)`

Reply:
This ain't my UI really. The code in OnCreate is  essentially cloning existing game UI to make sure that I have the same look, so the destroy here is to flush what I don't want to keep from the cloned element.

Reply:
Yeah, what I'm suggesting, create a `Panel container` gameobject with just a rect transform and shove all the child gameobjects there.

Then `Destroy(container)` will do everything for you, even if some _other_ code attaches gameobjects in it.

Reply:
Humm, not sure I see why it would be better. Because instead of doing:
- clone existing tab container
- destroy all gameObjects in it.

It would be:
- clone existing tab container
- create custom container
- iterate over all children of the tab container to set our own container as parent
- destroy the container

Reply:
Probably force of habit here, in case any other content needed to be cloned, you would generally need to care only about that one and Destroy it. But `YDNIN` applies here. Not needed.